sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/BlobBatch.java

```diff
@@ -131,7 +131,7 @@ public final class BlobBatch {
      * @throws UnsupportedOperationException If this batch has already added an operation of another type.
      */
     public Response<Void> deleteBlob(String containerName, String blobName) {
-        return deleteBlobHelper(containerName + "/" + Utility.urlEncode(Utility.urlDecode(blobName)), null, null);
+        return deleteBlobHelper(Utility.urlEncode(containerName) + "/" + Utility.urlEncode(blobName), null, null);
```

Comment:
Hmm that's unfortunate that we are already doing the url decoding and re-encoding here... I'm hesitant to change it at this point unless we are committed to updating this behavior, especially since we are trying to scope this change to container names